mm2src/mm2_test_helpers/src/for_tests.rs

```diff
@@ -2023,3 +2058,237 @@ pub async fn set_price(mm: &MarketMakerIt, base: &str, rel: &str, price: &str, v
+            "Trigger taker subscription to {}/{} orderbook topic first and sleep for 1 second",
+            base,
+            rel
+        );
+        let rc = taker
+            .rpc(&json!({
+                "userpass": taker.userpass,
+                "method": "orderbook",
+                "base": base,
+                "rel": rel,
+            }))
+            .await
+            .unwrap();
+        assert!(rc.0.is_success(), "!orderbook: {}", rc.1);
+        Timer::sleep(1.).await;
+        common::log::info!("Issue taker {}/{} buy request", base, rel);
+        let rc = taker
+            .rpc(&json!({
+                "userpass": taker.userpass,
+                "method": "buy",
+                "base": base,
+                "rel": rel,
+                "volume": volume,
+                "price": taker_price
+            }))
+            .await
+            .unwrap();
+        assert!(rc.0.is_success(), "!buy: {}", rc.1);
+        let buy_json: Json = serde_json::from_str(&rc.1).unwrap();
+        uuids.push(buy_json["result"]["uuid"].as_str().unwrap().to_owned());
+    }
+
+    for (base, rel) in pairs.iter() {
+        // ensure the swaps are started
+        let expected_log = format!("Entering the taker_swap_loop {}/{}", base, rel);
+        taker.wait_for_log(5., |log| log.contains(&expected_log)).await.unwrap();
+        let expected_log = format!("Entering the maker_swap_loop {}/{}", base, rel);
+        maker.wait_for_log(5., |log| log.contains(&expected_log)).await.unwrap()
+    }
+
+    uuids
+}
+
+pub async fn wait_for_swaps_finish_and_check_status(
+    maker: &mut MarketMakerIt,
+    taker: &mut MarketMakerIt,
+    uuids: &[impl AsRef<str>],
+    volume: f64,
+) {
+    for uuid in uuids.iter() {
+        maker
+            .wait_for_log(900., |log| {
+                log.contains(&format!("[swap uuid={}] Finished", uuid.as_ref()))
+            })
+            .await
+            .unwrap();
+
+        taker
+            .wait_for_log(900., |log| {
+                log.contains(&format!("[swap uuid={}] Finished", uuid.as_ref()))
+            })
+            .await
+            .unwrap();
+
+        log!("Waiting a few second for the fresh swap status to be saved..");
+        Timer::sleep(3.33).await;
+
+        log!("Checking taker status..");
+        check_my_swap_status(
+            taker,
+            uuid.as_ref(),
+            &TAKER_SUCCESS_EVENTS,
+            &TAKER_ERROR_EVENTS,
+            BigDecimal::try_from(volume).unwrap(),
+            BigDecimal::try_from(volume).unwrap(),
+        )
+        .await;
+
+        log!("Checking maker status..");
+        check_my_swap_status(
+            maker,
+            uuid.as_ref(),
+            &MAKER_SUCCESS_EVENTS,
+            &MAKER_ERROR_EVENTS,
+            BigDecimal::try_from(volume).unwrap(),
+            BigDecimal::try_from(volume).unwrap(),
+        )
+        .await;
+    }
+}
+
+pub async fn test_qrc20_history_impl(local_start: Option<LocalStart>) {
```

Comment:
I tried to run `cargo test --target wasm32-unknown-unknown --lib` and it attempted to compile tests of all workspace crates. It might be a way to go, but I prefer to do it later to avoid having even larger diff.

Reply:
Yes, this works but only partly because `cargo test --target wasm32-unknown-unknown --lib` runs some tests multiple times. For example, `indexed_db` tests are launched in almost all libraries that imports it, for example, from `coins`, `mm2_main`, `mm2_gui_storage` and other libraries.

Reply:
Wow, that's quite an overhead 🙂 I will recheck and give it a try later anyway.